nav2_waypoint_follower/src/waypoint_follower.cpp

```diff
@@ -64,6 +70,28 @@ WaypointFollower::on_configure(const rclcpp_lifecycle::State & /*state*/)
     get_node_waitables_interface(),
     "FollowWaypoints", std::bind(&WaypointFollower::followWaypoints, this), false);
 
+  if (task_executor_at_waypoint_id_ == default_task_executor_at_waypoint_id_) {
+    nav2_util::declare_parameter_if_not_declared(
+      this, default_task_executor_at_waypoint_id_ + ".plugin",
+      rclcpp::ParameterValue(default_task_executor_at_waypoint_type_));
+  }
+
+  try {
+    task_executor_at_waypoint_type_ = nav2_util::get_plugin_type_param(
+      this,
+      task_executor_at_waypoint_id_);
+    task_executor_at_waypoint_ = task_executor_at_waypoint_loader_.createUniqueInstance(
```

Comment:
`createUniqueInstance` returns a unique not a shared pointer, the `task_executor_at_waypoint_` variable is a shared pointer I believe